<!-- Launches: self_localization_selector_process -->
<launch>

  <!-- Basic data -->
  <arg name="drone_id_namespace"                    default="drone1" />
  <arg name="drone_id"                              default="1" />
  <arg name="my_stack_directory"                    default="$(env AEROSTACK_STACK)" />

  <!-- topics & services-->
  <arg name="change_mode_to_odometry_srv"           default="change_self_localization_mode_to_odometry" />
  <arg name="change_mode_to_visual_marker_srv"      default="change_self_localization_mode_to_visual_marker" />
  <arg name="odometry_estimated_pose_topic"         default="EstimatedPose_droneGMR_wrt_GFF" />
  <arg name="odometry_estimated_speed_topic"        default="EstimatedSpeed_droneGMR_wrt_GFF" />
  <arg name="aruco_slam_estimated_pose_topic"       default="ArucoSlam_EstimatedPose" />
  <arg name="aruco_slam_estimated_speed_topic"      default="ArucoSlam_EstimatedSpeed" />
  <arg name="estimated_pose_topic"                  default="estimated_pose" />
  <arg name="estimated_speed_topic"                 default="estimated_speed" />

  <group ns="$(arg drone_id_namespace)">
    <!-- Basic data -->
    <param name="drone_id_namespace"                value="$(arg drone_id_namespace)"             type="str" />
    <param name="drone_id"                          value="$(arg drone_id)"                       type="str" />
    <param name="my_stack_directory"                value="$(arg my_stack_directory)"             type="str" />

    <!-- topics & services -->
    <param name="~change_mode_to_odometry_srv"       value="$(arg change_mode_to_odometry_srv)"   type="str" />
    <param name="~change_mode_to_visual_marker_srv"  value="$(arg change_mode_to_visual_marker_srv)" type="str" />
    <param name="~odometry_estimated_pose_topic"     value="$(arg odometry_estimated_pose_topic)"  type="str" />
    <param name="~odometry_estimated_speed_topic"    value="$(arg odometry_estimated_speed_topic)" type="str" />
    <param name="~aruco_slam_estimated_pose_topic"   value="$(arg aruco_slam_estimated_pose_topic)" type="str" />
    <param name="~aruco_slam_estimated_speed_topic"  value="$(arg aruco_slam_estimated_speed_topic)" type="str" />
    <param name="~estimated_pose_topic"              value="$(arg estimated_pose_topic)"           type="str" />
    <param name="~estimated_speed_topic"             value="$(arg estimated_speed_topic)"          type="str" />

    <!-- node -->
    <node name="self_localization_selector_process" pkg="self_localization_selector_process"
	  type="self_localization_selector_process" output="screen"/>

    <!--<test test-name="BehaviorCoordinatorTest" pkg="behavior_coordinator_process" type="BehaviorCoordinatorTest"/>-->

  </group>

</launch>
